microservice-schedule/src/main/java/io/microprofile/showcase/schedule/persistence/ScheduleDAO.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2016 Microprofile.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.microprofile.showcase.schedule.persistence;
+
+import io.microprofile.showcase.schedule.model.Schedule;
+
+import javax.enterprise.context.ApplicationScoped;
+import java.util.*;
+
+@ApplicationScoped
+public class ScheduleDAO {
+
+    private long sequence = 0L;
+
+    private final Map<Long, Schedule> scheduleMap = new HashMap<>();
```

Comment:
OK, better to use thread safe data structures than synchronised blocks, but you can refactor that later